### PR TITLE
Add support for base64 images in image entity URLs

### PR DESCRIPTION
**Pull request by ksuprynowicz**
It supports the same standard URL scheme as glTF files for their base64 textures embedded in URLs.
Example URL:
```
data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAIAAAAC64paAAABhGlDQ1BJQ0MgcHJvZmlsZQAAKJF9kT1Iw0AcxV9TpSIVByuIOGSoDmKXKuKoVShChVArtOpgcukXNGlIUlwcBdeCgx+LVQcXZ10dXAVB8APE2cFJ0UVK/F9SaBHjwXE/3t173L0DhEaFaVbXLKDptplOJsRsblUMvSKMCAYRx7jMLGNOklLwHV/3CPD1Lsaz/M/9OfrUvMWAgEg8ywzTJt4gnt60Dc77xBFWklXic+IJky5I/Mh1xeM3zkWXBZ4ZMTPpeeIIsVjsYKWDWcnUiKeIo6qmU76Q9VjlvMVZq9RY6578heG8vrLMdZojSGIRS5AgQkENZVRgI0arToqFNO0nfPzDrl8il0KuMhg5FlCFBtn1g//B726twmTcSwongO4Xx/kYBUK7QLPuON/HjtM8AYLPwJXe9lcbwMwn6fW2Fj0C+reBi+u2puwBlzvA0JMhm7IrBWkKhQLwfkbflAMGboHeNa+31j5OH4AMdZW6AQ4OgbEiZa/7vLuns7d/z7T6+wEIF3LiWsAO7gAAAAlwSFlzAAAuIwAALiMBeKU/dgAAAAd0SU1FB+gMFBU6ABMMoWQAAAAZdEVYdENvbW1lbnQAQ3JlYXRlZCB3aXRoIEdJTVBXgQ4XAAAATUlEQVQ4y2P8//8/A9kAWTORbDhggsvBVUAk0NhoaiCACdMtjIyMyGxkLhpgwuMRZCdgBSy4bP7//z8eO6EqKQltJgYKwKjmUc3DVTMAN4JB9I3iT2YAAAAASUVORK5CYII=
```
overte-snap-by-X74hc595-on-2024-12-20_23-23-01

For local entities bigger files are supported too:

overte-snap-by-X74hc595-on-2024-12-20_23-43-30

### CODE REVIEW

File: libraries/material-networking/src/material-networking/TextureCache.cpp [hunk 1]
```diff
@@ -254,6 +254,18 @@ NetworkTexturePointer TextureCache::getTexture(const QUrl& url, image::TextureUs
     if (url.scheme() == RESOURCE_SCHEME) {
         return getResourceTexture(url);
     }
+
+    QString urlString = url.toString();
+    if (content.isEmpty() && (urlString.contains("data:image/jpeg;base64,")
```
Comment (HifiExperiments) [inline libraries/material-networking/src/material-networking/TextureCache.cpp:259]: pretty cool 😁

should these “contains” be “startsWith”?

**Comment by Armored-Dragon** [comment]
This will work for what I want! Out of curiosity, what happens when you try and send a image that is too big over the network? Will the image just not load / show up corrupted?

File: libraries/material-networking/src/material-networking/TextureCache.cpp (hunk 1 above)
Comment (ksuprynowicz) [reply libraries/material-networking/src/material-networking/TextureCache.cpp]: Thank you! I fixed this and also added documentation in Image entity's imageURL property.

**Review by HifiExperiments — APPROVED** [review]

**Comment by HifiExperiments** [comment]
> This will work for what I want! Out of curiosity, what happens when you try and send a image that is too big over the network? Will the image just not load / show up corrupted?

they’ll render fine locally, but won’t be transmitted over the network to other people.  which means they also won’t make it to the server to be saved.  but they’d be fine for local entities like 74 said

**Pull request merged**